src/libopensc/gp.h

```diff
@@ -32,9 +32,34 @@
 extern "C" {
 #endif
 
+/* returned by the iso get status apdu with the global platform cplc data parameters */
+typedef struct global_platform_cplc_data {
+	u8 tag[2];
+	u8 length;
+	u8 ic_fabricator[2];
+	u8 ic_type[2];
+	u8 os_id[2];
+	u8 os_date[2];
+	u8 os_level[2];
+	u8 fabrication_data[2];
+	u8 ic_serial_number[4];
+	u8 ic_batch[2];
+	u8 module_fabricator[2];
+	u8 packing_data[2];
+	u8 icc_manufacturer[2];
+	u8 ic_embedding_data[2];
+	u8 pre_personalizer[2];
+	u8 ic_pre_personalization_data[2];
+	u8 ic_pre_personalization_id[4];
+	u8 ic_personalizaer[2];
+	u8 ic_personalization_data[2];
+	u8 ic_personalization_id[4];
+} global_platform_cplc_data_t;
```

Comment:
the attributes need to be packed, right?

Reply:
Ideally, this should return the interpreted data only, i.e. without the ASN.1 overhead...

Reply:
Yes. The data is read directly into it from the response in APDU. So I believe it would be more safe to define that packet. I will add it in following commits (especially as it is not completely related to the issue that I was fixed).

Reply:
I don't see this doing anything with ASN.1.

Reply:
What about tag and length? do they nominate something different than an ASN.1 header?

Reply:
Ok, it is `BER-TLV` response from the GET DATA instruction, but since this one has pretty strict encoding (2B tag, 1B length), I do not think explicit parsing using ASN1 library is needed here. I would just check the length (already done), tag and internal length. See the last commit (rebased on master).

> When the value of P1-P2 lies in the range from ‘4000’ to ‘FFFF’, the value of P1-P2 shall be a BER-TLV tag on two bytes.